### DISCUSSION TITLE
Default Media Sorting improvements

### DISCUSSION THREAD

**Discussion by thenexus00** [umbraco/Umbraco-CMS#16766] [Features and ideas]
**1. Folders displayed first:**
AS with most? file systems folder items should always be first displayed then files regardless of sort order

**2. Media Library default sort and display options:**
The media library needs default sort options. I find it annoying that new files are added last so it would be great to have options like date asc, date dec along with name and type sorting options to be able to set as default.

**3. Media picker display options:**
As with the media Library if you have a media picker you build in a project that represents files the list view to view file names is a preferred layout. If it is an image then the grid would be best but as it stands it defaults to grid. It would be good to be able to choose the default here to represent your use case. The sorting as well like in 2 should be an option. Along with the possible bug noted in 4 it is annoying to add a file in a picker and then have to scroll all the way down to select it.

**4. Media Item bug fix? (Upload item)**
Pretty sure a long while ago if you uploaded a single file it would auto select it. This does not occur any more.

**Comment by psot** [comment]
Working with folders is quite frustrating because every user expects to find the folders on top.
Sorting and auto-sorting on folder contents is also a must and this should be a higher priority with Umbraco 15 since manually sorting the folder contents, only sorts them in the tree view and not in the collection view!
This should be a fairly easy implementation.
I'd also add a physical filename rename function in the Media section as a suggestion.

**Comment by tigreye007** [comment]
This is quite important, and a reason some of our clients second guess why we would choose this CMS over others, when what they see as an "expected" experience like this doesn't exist.  It looks bad on **_US_** of course once they see how this functions.  I do see that if you choose the "List View" view, you can then sort by name by clicking the column header, but ideally this would be available in Grid view as well, and would also allow you to somehow default both of these views to always be alphabetical (with folders first).

If this is too difficult to manage, one "quick fix" option would be to allow a user, within the "sort children" views, to click the Name header like you're able to do when sorting within the Content tree.  However, this would only be a stop gap solution.
